plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,363 @@
+	resourceToServicesMap := make(map[string][]transit.DynamicMonitoredService)
+	for _, metric := range metrics {
+		resource, service := g.parseMetric(metric)
+		resourceToServicesMap[resource] = append(resourceToServicesMap[resource], service)
+	}
+
+	var resources []transit.DynamicMonitoredResource
+	for resourceName, services := range resourceToServicesMap {
+		resources = append(resources, transit.DynamicMonitoredResource{
+			BaseResource: transit.BaseResource{
+				BaseTransitData: transit.BaseTransitData{
+					Name: resourceName,
+					Type: transit.Host,
+				},
+			},
+			Status:        transit.HostUp,
+			LastCheckTime: transit.NewTimestamp(),
+			Services:      services,
+		})
+	}
+
+	traceToken, _ := uuid.GenerateUUID()
+	requestJSON, err := json.Marshal(transit.DynamicResourcesWithServicesRequest{
+		Context: &transit.TracerContext{
+			AppType:    "TELEGRAF",
+			AgentID:    g.AgentID,
+			TraceToken: traceToken,
+			TimeStamp:  transit.NewTimestamp(),
+			Version:    transit.ModelVersion,
+		},
+		Resources: resources,
+		Groups:    nil,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	headers := map[string]string{
+		"GWOS-APP-NAME":  "telegraf",
+		"GWOS-API-TOKEN": g.authToken,
+		"Content-Type":   "application/json",
+		"Accept":         "application/json",
+		"User-Agent":     internal.ProductToken(),
+	}
+
+	statusCode, body, err := clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+	if err != nil {
+		return fmt.Errorf("error while sending: %v", err)
+	}
+
+	/* Re-login mechanism */
+	if statusCode == 401 {
+		if err = g.Connect(); err != nil {
+			return fmt.Errorf("re-login failed: %v", err)
+		}
+		headers["GWOS-API-TOKEN"] = g.authToken
+		statusCode, body, err = clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+		if err != nil {
+			return fmt.Errorf("error while sending: %v", err)
+		}
+	}
+
+	if statusCode != 200 {
+		return fmt.Errorf("HTTP request failed. [Status code]: %d, [Response]: %s", statusCode, string(body))
+	}
+
+	return nil
+}
+
+func (g *Groundwork) Description() string {
+	return "Send telegraf metrics to GroundWork Monitor"
+}
+
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{
+			ResourceTag:         "host",
+			DefaultHost:         "telegraf",
+			DefaultServiceState: string(transit.ServiceOk),
+		}
+	})
+}
+
+func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := g.DefaultHost
+	if value, present := metric.GetTag(g.ResourceTag); present {
+		resource = value
+	}
+
+	service := metric.Name()
+	if value, present := metric.GetTag("service"); present {
+		service = value
+	}
+
+	status := g.DefaultServiceState
+	value, statusPresent := metric.GetTag("status")
+	if validStatus(value) {
+		status = value
+	}
+
+	message := ""
+	if value, present := metric.GetTag("message"); present {
+		message = value
+	}
```

Comment:
As metric always return a string (even if not found) you can do
```suggestion
	message, _ := metric.GetTag("message")
```

Reply:
Done